mgmtsystem/google.py

```diff
@@ -127,9 +128,17 @@ def list_image(self):
 
     def _find_instance_by_name(self, instance_name):
         try:
-            instance = self._instances.get(
-                project=self._project, zone=self._zone, instance=instance_name).execute()
-            return instance
+            # by default search the instances in current zone..
+            for instance in self._get_zone_instances(self._zone).get('items', []):
```

Comment:
Excellent, however, I think the previous first call here is better, to use. Why get all and then do the for loop? If you request a specific one it should be faster, you can then catch the exception if it's not found and run the second step. This will always be quicker as you are a) searching at source, and b) transferring less data across the wire.